soh/soh/OTRGlobals.cpp

```diff
@@ -925,6 +927,84 @@ bool PathTestCleanup(FILE* tfile) {
+    // Add backslashes.
+    for (auto i = stringValue.begin();;) {
+        auto const pos = std::find_if(i, stringValue.end(), [](char const c) { return '\\' == c || '\'' == c || '"' == c; });
+        if (pos == stringValue.end()) {
+            break;
+        }
+        i = std::next(stringValue.insert(pos, '\\'), 2);
+    }
+
+    // Removes others.
+    stringValue.erase(std::remove_if(stringValue.begin(), stringValue.end(), [](char const c) {
+        return '\n' == c || '\r' == c || '\0' == c || '\x1A' == c; }), stringValue.end());
+
+    return stringValue;
+}
+
+static const char* lastDroppedFile = "";
+
+void processDroppedFiles() {
+    const char* droppedFile = CVarGetString("gDroppedFile", "");
+    if (strcmp(droppedFile, lastDroppedFile) != 0 && strcmp(droppedFile, "") != 0) {
+        lastDroppedFile = droppedFile;
+        try {
+            std::ifstream configStream(sanitizePath(droppedFile));
+            if (!configStream) {
+                return;
+            }
+
+            nlohmann::json configJson;
+            configStream >> configJson;
+
+            if (!configJson.contains("CVars")) {
+                return;
+            }
+
+            clearCvars(enhancementsCvars);
+            clearCvars(cheatCvars);
+            clearCvars(randomizerCvars);
+
+            // Flatten everything under CVars into a single array
+            auto cvars = configJson["CVars"].flatten();
+
+            for (auto& [key, value] : cvars.items()) {
+                // Replace slashes with dots in key, and remove leading dot
+                std::string path = key;
+                std::replace(path.begin(), path.end(), '/', '.');
+                if (path[0] == '.') {
+                    path.erase(0, 1);
+                }
+                if (value.is_string()) {
+                    CVarSetString(path.c_str(), value.get<std::string>().c_str());
+                } else if (value.is_number_integer()) {
+                    CVarSetInteger(path.c_str(), value.get<int>());
+                } else if (value.is_number_float()) {
+                    CVarSetFloat(path.c_str(), value.get<float>());
+                }
+            }
+
+            auto gui = LUS::Context::GetInstance()->GetWindow()->GetGui();
+            gui->GetGuiWindow("Console")->Hide();
+            gui->GetGuiWindow("Actor Viewer")->Hide();
+            gui->GetGuiWindow("Collision Viewer")->Hide();
+            gui->GetGuiWindow("Save Editor")->Hide();
+            gui->GetGuiWindow("Display List Viewer")->Hide();
+            std::reinterpret_pointer_cast<LUS::ConsoleWindow>(LUS::Context::GetInstance()->GetWindow()->GetGui()->GetGuiWindow("Console"))->ClearBindings();
+
+            gui->SaveConsoleVariablesOnNextTick();
+
+            uint32_t finalHash = boost::hash_32<std::string>{}(configJson.dump());
+            gui->GetGameOverlay()->TextDrawNotification(30.0f, true, "Configuration Loaded. Hash: %d", finalHash);
+        } catch (...) {
```

Comment:
pokemon exception handling with no logging or anything here seems less than ideal

Reply:
How's that? https://github.com/HarbourMasters/Shipwright/compare/9de168c259cb91f6870870616cbfd7293d29deb5..45e5004c181c60e5e2897d5ac128c0b0017de7fa

Reply:
beautiful